audio-worklet/src/noisegate-script-processor.js

```diff
@@ -40,39 +52,39 @@ class NoiseGate {
 
     this.context_ = context;
     const bufferSize = options.bufferSize || 0;
-    // The default threshold is 140 decibels (loudest sound a human can hear
-    // without pain) so that the noise gate does not mute signals by default.
-    this.threshold = options.threshold || 140;
+    this.threshold = options.threshold || 0;
     this.attack = options.attack || 0;
     this.release = options.release || 0;
+    
+    // The time constant of the filter has been set experimentally to minimize
+    // delay while still adequately suppressing high frequency oscillation.
     const timeConstant = options.timeConstant || 0.0025;
 
     // Alpha controls a tradeoff between the smoothness of the
     // envelope and its delay, with a higher value giving more smoothness at
-    // the expense of delay and vice versa. The time constant of the filter
-    // has been set experimentally to minimize delay while still adequately
-    // suppressing high frequency oscillation.
-    this.alpha_ = this.getFilterCoefficient_(timeConstant);
-   
-    this.node_ = this.context_.createScriptProcessor(
+    // the expense of delay and vice versa.
+    this.alpha_ = NoiseGate.getAlphaFromTimeConstant_(
+        timeConstant, this.context_.sampleRate);
+
+    this.noiseGateKernel_ = this.context_.createScriptProcessor(
         bufferSize, numberOfChannels, numberOfChannels);
-    this.node_.onaudioprocess = this.onaudioprocess_.bind(this);
+    this.noiseGateKernel_.onaudioprocess = this.onaudioprocess_.bind(this);
 
     // The noise gate is connected to and from by dummy input and output nodes.
     this.input = new GainNode(this.context_);
     this.output = new GainNode(this.context_);
-    this.input.connect(this.node_).connect(this.output);
+    this.input.connect(this.noiseGateKernel_).connect(this.output);
 
-    // The last envelope level of a given buffer.
-    this.lastLevel_ = 0;
+    // The previous envelope level, a float representing signal amplitude.
+    this.previousLevel_ = 0;
 
     // The last weight (between 0 and 1) assigned, where 1 means the gate
     // is open and 0 means it is closed and the sample in the output buffer is
     // muted.
```

Comment:
This comment isn't really quite complete if you're doing attack and release, right?  And it's really the memory needed to implement attack and release.  Because if you don't have that, this isn't necessary, I think.